conformance/packages/conformance-tests/src/resolver/dnssec/scenarios/bogus.rs

```diff
@@ -513,3 +517,142 @@ AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="
 
     Ok(())
 }
+
+#[test]
+fn unauthenticated_nsec_wildcard_name() -> Result<()> {
+    let wildcard_fqdn = FQDN::EXAMPLE_SUBDOMAIN.push_label("*");
+    let any_modified = AtomicBool::new(false);
+    invalid_nsec_wildcard_no_data_test(&|zone, records| {
+        if *zone == FQDN::TEST_DOMAIN {
+            for record in records.iter_mut() {
+                if let Record::RRSIG(rrsig) = record {
+                    if rrsig.fqdn == wildcard_fqdn && rrsig.type_covered == RecordType::NSEC {
+                        rrsig.signature = "AAAA".to_owned();
+                        any_modified.store(true, Ordering::SeqCst);
+                    }
+                }
+            }
+        }
```

Comment:
I think you can make this less deeply nested by writing it like:

```rust
        if *zone != FQDN::TEST_DOMAIN {
            return;
        }
        for rrsig in records.iter_mut().filter_map(Record::as_rrsig_mut) {
            if rrsig.fqdn == wildcard_fqdn && rrsig.type_covered == RecordType::NSEC {
                rrsig.signature = "AAAA".to_owned();
                any_modified.store(true, Ordering::SeqCst);
            }
        }
```

If you like that suggestion it will also apply to a few other tests added in this branch. Some of them can benefit from both the early return & the switch to `filter_map()` while the ones using `retain()` would only benefit from the early return for the `zone` check.